File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def digit(x):
            pr = 1
            
            while x:
                rem = x % 10
                pr *= rem
                x //= 10
                
            return pr

        while True:
            if digit(n) % t == 0:
                return n
            
            n += 1

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.go
func smallestNumber(n int, t int) int {
	for {
		x := n
		prod := 1

		for x > 0 {
			prod *= x % 10
			x /= 10
		}

		if prod%t == 0 {
			return n
		}

		n++
	}
}